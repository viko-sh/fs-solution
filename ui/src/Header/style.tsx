import styled from 'styled-components';

export const HeaderWrapper = styled.div`
  border: 1px solid #c7c7c7;
  background: #fff;
  margin: 30px 0 0 0;
  box-shadow: 0 0 5px rgba(199, 199, 199, 0.4);
`;

export const Tabs = styled.div`
  display: flex;
  flex-direction: row;
`;

export const Tab = styled.div`
  display: block;
  padding: 0 0;
  flex: 1;
  text-align: center;
  align-items: center;
  font-size: 18px;
  cursor: pointer;

  a {
    width: 100%;
    display: block;
    padding: 15px 0;
    text-decoration: none;
    color: rgb(85, 85, 85);
  }

  &:first-of-type {
    border-right: 1px solid #c7c7c7;
  }
  &.selected {
    color: #000;
    font-weight: bold;
    border-bottom: 3px #ff7a12 solid;
  }
`;
