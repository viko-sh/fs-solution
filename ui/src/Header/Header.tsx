import React, { Component } from 'react';
import { HeaderWrapper, Tabs, Tab } from './style';
import { Link, withRouter } from 'react-router-dom';

export class HeaderInner extends Component<any> {
  render() {
    const {
      location: { pathname }
    } = this.props;

    return (
      <HeaderWrapper>
        <Tabs>
          <Tab className={pathname === '/' ? 'selected' : ''}>
            <Link to="/">Invited</Link>
          </Tab>
          <Tab className={pathname === '/accepted' ? 'selected' : ''}>
            <Link to="/accepted">Accepted</Link>
          </Tab>
        </Tabs>
      </HeaderWrapper>
    );
  }
}

export const Header = withRouter(HeaderInner);
