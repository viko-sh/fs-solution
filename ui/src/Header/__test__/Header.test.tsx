import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { HeaderInner } from '../Header';

const defaultProps = {
  location: {
    pathname: '/accepted'
  }
};

const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = shallow(<HeaderInner {...props} />);
  return { props, wrapper };
};

describe('Header component', () => {
  it('Should render header', done => {
    const { wrapper } = mountComponent();
    expect(wrapper).toHaveLength(1);
    done();
  });
  it('should be match with snapshot', done => {
    // Arrange (no Act required)
    const { wrapper } = mountComponent();
    // Assert
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
