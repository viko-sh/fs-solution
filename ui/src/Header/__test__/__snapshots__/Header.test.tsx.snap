// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be match with snapshot 1`] = `ShallowWrapper {}`;
